cluster/gce/gci/flexvolume_node_setup.sh

```diff
@@ -99,31 +99,31 @@ generate_chroot_wrapper() {
     exit 1
   fi
 
-  for driver_dir in ${MOUNTER_PATH}/flexvolume/*; do
+  for driver_dir in "${MOUNTER_PATH}/flexvolume"/*; do
     if [ -d "$driver_dir" ]; then
 
-      filecount=$(ls -1 $driver_dir | wc -l)
-      if [ $filecount -gt 1 ]; then
+      filecount=$(find "$driver_dir" -mindepth 1 -maxdepth 1 -printf "%P\n" | grep -c -v "^\.")
+      if [ "$filecount" -gt 1 ]; then
         echo "ERROR: Expected 1 file in the FlexVolume directory but found $filecount."
         exit 1
       fi
 
-      driver_file=$( ls $driver_dir | head -n 1 )
+      driver_file=$(find "$driver_dir" -mindepth 1 -maxdepth 1 -printf "%P\n" | grep -v "^\." | head -n 1)
```

Comment:
```
driver_file=$( ls $driver_dir | head -n 1 )
               ^------------^ SC2012: Use find instead of ls to better handle non-alphanumeric filenames.
                  ^---------^ SC2086: Double quote to prevent globbing and word splitting.
```

Reply:
same here, I think it's acceptable to ignore SC2012

Reply:
instead of printf, use normal print, and use basename as the last step in the chain

Reply:
I have done.